src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/LinkerTests.cs

```diff
@@ -289,11 +289,16 @@ public AttributedButtonStub (Context context) : base (context)
 			}
 		}
 
+		// mode
+		//   0 .. Debug configuration
+		//   1 .. Debug configuration, AndroidAddKeepAlives=true
+		//   2 .. Release configuration
+		//   3 .. Release configuration, AndroidLinkMode=None
 		[Test]
-		public void AndroidAddKeepAlives ()
+		public void AndroidAddKeepAlives ([Values (0, 1, 2, 3)] int mode)
```

Comment:
I would prefer:

```csharp
public void AndroidAddKeepAlives ([Values ("Debug", "Release")] string configuration, [Values ("", "True")] string keepAlives, [Values ("", "None")] string linkMode)
```

This reduces my mental requirements -- no "magic numbers" -- and also builds the combinatorial for all those properties.

Reply:
Do we really want to test 16 combinations instead of 4 important ones?

Reply:
If you want to only have 4 configurations then you should use `TestCaseSource`

```
#pragma warning disable 414
    static object [] KeepAlivesTestCases = new object [] {
        new object [] {
             /* configuration*/ "Release",
             /* keepAlives */ "True",
            /* linkMode */ "",
        },
        new object [] {
             /* configuration*/ "Release",
             /* keepAlives */ "False",
            /* linkMode */ "",
        }
    }
#pragma warning restore 414
    [Test]
    [TestCaseSource(nameof(KeepAlivesTestCases))]
    public void AndroidAddKeepAlives (string configuration, string keepAlives, string linkMode)
```

This is a common pattern in our unit tests https://github.com/xamarin/xamarin-android/blob/7b51fd1b45f7f315df402ba333e141a49c8a3ca7/tests/MSBuildDeviceIntegration/Tests/DebuggingTest.cs#L120